docs/handler-type-comparison.md

```diff
@@ -0,0 +1,131 @@
+# Handler type comparison
+
+This document compares the four Dogma handler types, highlighting their roles in
+message flow, data access, and external interaction.
+
+If you're new to Dogma, start by familiarising yourself with the key [concepts].
+
+> [!TIP]
+> Consult the [glossary] for definitions of terms used in this document.
+
+## Comparison matrix
+
+|                        | Aggregate | Process  | Integration |  Projection  |
+| ---------------------- | :-------: | :------: | :---------: | :----------: |
+| Command messages       |     ↓     |    ↑     |      ↓      |              |
+| Event messages         |     ↑     |    ↓     |      ↑      |      ↓       |
+| Timeout messages       |           |    ⇅     |             |              |
+| [State]                |     ✓     |    ✓     |             |              |
+| [External data access] |           |    ~     |      ✓      |      ✓       |
+| [Isolation boundary]   | instance  | instance |   command   | event-stream |
+| [Event stream binding] | instance  |    —     |   command   |      —       |
+
+#### Legend
+
+- `✓` supported
+- `~` supported, but not recommended
+- `↓` consumes
+- `↑` produces
+- `⇅` produces & consumes
+- `—` not applicable
+
+## Definitions
+
+### State
+
+Aggregates and processes maintain state across many separate instances.
+The engine routes each message to a specific instance using an identifier
+provided by the application.
+
+Each aggregate instance records events, and each process instance holds data.
```

Comment:
This seems slightly off somehow with how it separates aggregate and instance for holding data?

Reply:
I was trying to say that the application's state is represented by the events an aggregate records, and the data inside a _process_ instance, but I also forgot about the events produced by integrations.

Reply:
I've reworded this section in an attempt to make it clearer. I also updated the glossary definition of "state" to make it clear that only events from an aggregate contribute to "state", per dogma's definition.